Matrix.SynapseInterop.Replication/DataRows/FederationStreamRow.cs

```diff
@@ -0,0 +1,66 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using Matrix.SynapseInterop.Replication.Structures;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+
+namespace Matrix.SynapseInterop.Replication.DataRows
+{
+    public class FederationStreamRow : IReplicationDataRow
+    {
+        public Dictionary<string[], EduEvent> keyedEdus;
+        public List<EduEvent> edus;
+        public List<string> devices;
+        public List<PresenceState> presence;
+
+        private FederationStreamRow()
+        {
+            keyedEdus = new Dictionary<string[], EduEvent>();
+            edus = new List<EduEvent>();
+            devices = new List<string>();
+            presence = new List<PresenceState>();
```

Comment:
lint that doesn't exist: assign variables where they are defined, not in the constructor